src/sidebar/components/ShareDialog/test/ImportAnnotations-test.js

```diff
@@ -0,0 +1,225 @@
+import { mount } from 'enzyme';
+
+import { checkAccessibility } from '../../../../test-util/accessibility';
+import { waitFor, waitForElement } from '../../../../test-util/wait';
+import ImportAnnotations, { $imports } from '../ImportAnnotations';
+
+describe('ImportAnnotations', () => {
+  let fakeImportAnnotationsService;
+  let fakeStore;
+
+  beforeEach(() => {
+    fakeImportAnnotationsService = {
+      import: sinon.stub().resolves([]),
+    };
+
+    fakeStore = {
+      profile: sinon.stub().returns({ userid: 'acct:john@example.com' }),
+      defaultAuthority: sinon.stub().returns('example.com'),
+      isFeatureEnabled: sinon.stub().returns(true),
+      importsPending: sinon.stub().returns(0),
+    };
+
+    $imports.$mock({
+      '../../store': { useSidebarStore: () => fakeStore },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  function createImportAnnotations() {
+    return mount(
+      <ImportAnnotations
+        store={fakeStore}
+        importAnnotationsService={fakeImportAnnotationsService}
+      />,
+    );
+  }
+
+  it('shows a notice if the user is not logged in', () => {
+    fakeStore.profile.returns({ userid: null });
+    const wrapper = createImportAnnotations();
+    assert.isTrue(wrapper.exists('[data-testid="log-in-message"]'));
+    assert.isFalse(wrapper.exists('input[type="file"]'));
+  });
+
+  function getImportButton(wrapper) {
+    return wrapper.find('button[data-testid="import-button"]');
+  }
+
+  function selectFile(wrapper, data) {
+    const fileInput = wrapper.find('input[type="file"]');
+    const fileContent = typeof data === 'string' ? data : JSON.stringify(data);
+    const file = new File([fileContent], 'export.json');
+
+    // `HTMLInputElement.files` can be assigned, but is a `FileList`, which
+    // can't be constructed, so we just stub the property instead.
+    Object.defineProperty(fileInput.getDOMNode(), 'files', {
+      get: () => [file],
+    });
+
+    fileInput.simulate('change');
+  }
```

Comment:
> Perhaps we can place all helper functions before any actual tests (getImportButton, selectFile, etc), even if they are not used by the first one. 

Yes, it probably makes sense to do this. I would suggest that the guideline would be "at the top of the innermost `describe` block containing tests that use the helper" rather than "at the top of the file".